New stuff

DRSchlaubi opened the pull request:
See #154, #156 
This redactors a lot of the bots core so I might sum it up a bit here:

- Use Kord instead of JDA (Fixes #156)
- Use ktor as the http Client because it provides a coroutine based API which okhttp doesn't
- Use [konf](https://github.com/uchuhimo/konf) instead of plain dotenv because the way we did it before was horrible (we still use dotenv for the config so the .env file remains and the old config only needs minor changes but we have a spec for our config now so automatic checking of types and presence of required types)
- Use slash commands (:eyes:)
- replace Autohelp with [the new Autohelp](https://github.com/bankobotv14/BankoBot/tree/main/autohelp) because the old ~~one is bad~~ okay let's just say it's better than the old one and not hate my old work too much (Fix #147, #106)

DRSchlaubi committed:
Initial pass on slash commands
- Migrate command system to slash commands
- Use interaction followups and responses instead of channel messages
- Remove everything related to javadocs

DRSchlaubi committed:
Migrate permission system to slash commands

DRSchlaubi committed:
A lot of fixes
- Fix sub commands
- Remove no longer needed command API parts
- Remove outdated tests

DRSchlaubi committed:
Optimize message sending code

DRSchlaubi committed:
Fix commands that take multi-line arguments

DRSchlaubi committed:
Fix commands that take multi-line arguments

DRSchlaubi committed:
Remove AutoHelp
- Remove dependencies from jcenter()

DRSchlaubi committed:
Update other dependencies

DRSchlaubi committed:
Switch to ktor-client to have a coroutine based http client API
- Make DevcordBot implement CoroutineScope

DRSchlaubi committed:
Fix Rest request serialization

DRSchlaubi committed:
Initial refactor iteration
- Migrate to Kord
- refactor slash command creation
- Migrate remaining rest request to ktor
- Various Code optimizations
- Remove custom JDA permission handlers

DRSchlaubi committed:
Migrate RatProtector.kt

DRSchlaubi committed:
Make Exposed understand Snowflakes

DRSchlaubi committed:
Improve configuration system

DRSchlaubi committed:
Update docs

DRSchlaubi committed:
Fix compiler error

DRSchlaubi committed:
Fix config issues

DRSchlaubi committed:
Fix command registration

DRSchlaubi commented:
~~Friendly reminder to add the GUILD_MEMBERS intent for member join event~~
**edit**: done

DRSchlaubi committed:
Add GUILD_MEMBERS intent

DRSchlaubi committed:
Merge remote-tracking branch 'origin/develop' into feature/refactor2

# Conflicts:
#	src/main/kotlin/com/github/devcordde/devcordbot/Launcher.kt
#	src/main/kotlin/com/github/devcordde/devcordbot/commands/general/TagCommand.kt
#	src/main/kotlin/com/github/devcordde/devcordbot/core/autohelp/ContentFetcher.kt

DRSchlaubi commented:
Oh wow that's less conflicts than I expected

DRSchlaubi committed:
Add the new autohelp
- Add gradle notation config

DRSchlaubi committed:
Merge remote-tracking branch 'origin/develop' into feature/refactor2

# Conflicts:
#	.github/workflows/gradle.yml

DRSchlaubi committed:
Disable game animation due to bugs

daspositioned commented on .env.example:
```diff
@@ -1,14 +1,13 @@
-DATABASE="database_name"
-DATABASE_USERNAME="user"
-DATABASE_PASSWORD="a_super_safe_password"
+DISCORD_GUILD_ID=""
+DISCORD_GAMES="separated;games"
+DATABASE_HOST="173.212.241.148"
+DATABASE_NAME=""
+DATABASE_USERNAME=""
+DATABASE_PASSWORD=""
```
Why exactly are the example values empty not? And what's that IP?

DRSchlaubi replied:
because I copied my actual config over and just removed they keys and forogt about the ip

daspositioned replied:
Alright, but please use the previous or at least similar values again

daspositioned commented on build.gradle.kts:
```diff
@@ -80,27 +89,17 @@ dependencies {
-        archiveClassifier.value = "original"
+    withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile> {
+        kotlinOptions {
+            jvmTarget = "14"
```
Is 15/16 not supported yet?

DRSchlaubi replied:
it is since 4 a few hrs ago as Kotlin 1.5 released and my java 15 jdk had issues. But the 1.5 gradle plugin is not in the Gradle repos yet so we need to wait for that

DRSchlaubi replied:
I upgraded it to Kotlin 1.5 and targetted java 16

daspositioned commented on src/main/java/de/nycode/bankobot/docdex/FlexmarkUtils.java:
```diff
@@ -0,0 +1,121 @@
+ *
+ */
+
+package de.nycode.bankobot.docdex;
```
Why is this not in the default package? And why Java?

DRSchlaubi replied:
> Why is this not in the default package? 

Because that code was copied from another project to give them credit license and package remained similar to how it`s done with other projects.

>  And why Java?

Because writing flexmark code in kotlin is horrible

daspositioned commented on src/main/java/net/dv8tion/jda/api/utils/MarkdownSanitizer.java:
```diff
@@ -0,0 +1,487 @@
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.api.utils;
```
Same here, why is this not in our package, and why Java?

DRSchlaubi replied:
same as above

daspositioned commented on src/main/kotlin/com/github/devcordde/devcordbot/Launcher.kt:
```diff
@@ -60,38 +55,20 @@ fun main(args: Array<String>) {
+
     if (debugMode) {
-        Sentry.init() // Initilizing sentry with null does mute sentry
+        Sentry.init("") // Initilizing sentry with null does mute sentry
```
That's not null

DRSchlaubi replied:
It's not because null no longer does what it did before as Sentry changed their client

daspositioned replied:
Obviously yes, but remove the comment then

DRSchlaubi replied:
I am not going to remove it but i will update it to say empty string instead

daspositioned commented on src/main/kotlin/com/github/devcordde/devcordbot/command/CommandClient.kt:
```diff
@@ -16,6 +16,10 @@
 
+import dev.kord.core.Kord
+import dev.kord.core.on
+import dev.kord.core.event.interaction.InteractionCreateEvent
```
Unused import?

DRSchlaubi replied:
kt-lint flags unused imports and it doesn't flag this

daspositioned commented on src/main/kotlin/com/github/devcordde/devcordbot/command/RegisterableCommand.kt:
```diff
@@ -0,0 +1,29 @@
+     * Registers the command to this [ApplicationCommandsCreateBuilder].
+     */
+    fun ApplicationCommandsCreateBuilder.applyCommand()
+}
```
Missing newline

daspositioned commented on src/main/kotlin/com/github/devcordde/devcordbot/command/context/Context.kt:
```diff
@@ -18,140 +18,138 @@ package com.github.devcordde.devcordbot.command.context
      */
-    val messageId: Long
-        get() = message.idLong
+    val jda: Kord
```
I don't think jda is the right variable name here

DRSchlaubi replied:
it's in fact not

daspositioned replied:
Uhm?

daspositioned commented on src/main/kotlin/com/github/devcordde/devcordbot/command/impl/CommandClientImpl.kt:
```diff
@@ -64,111 +62,82 @@ class CommandClientImpl(
         )
 
+        val arguments = Arguments(interaction.command.options)
+        val ack = interaction.ackowledgePublic()
```
Don't use such abbreviations, we don't use them anywhere else either

daspositioned commented on src/main/kotlin/com/github/devcordde/devcordbot/command/impl/DebugErrorHandler.kt:
```diff
@@ -35,7 +36,8 @@ class DebugErrorHandler : ErrorHandler {
+            context.respond("Es ist ein Fehler aufgetreten! Der ErrorHandler wurde wegen des Debug modus deaktiviert")
+        }
     }
 }
```
Missing newline

daspositioned commented on src/main/kotlin/com/github/devcordde/devcordbot/commands/owners/EvalCommand.kt:
```diff
@@ -64,40 +68,36 @@ class EvalCommand : AbstractCommand() {
                 import net.dv8tion.jda.api.*
                 import net.dv8tion.jda.api.entities.*
```
Update these imports

daspositioned commented on src/main/kotlin/com/github/devcordde/devcordbot/config/ConfigLoader.kt:
```diff
@@ -0,0 +1,76 @@
+}
+
+private inline fun <reified T> SimpleModule.addDeserializer(deserializer: JsonDeserializer<T>) =
+    addDeserializer(T::class.java, deserializer)
```
Missing newline

daspositioned commented on src/main/kotlin/com/github/devcordde/devcordbot/config/ConfigSpec.kt:
```diff
@@ -0,0 +1,87 @@
+import com.uchuhimo.konf.ConfigSpec as KonfigSpec
+
+internal object ConfigSpec : KonfigSpec("") {
+    val xpWhielist by optional(SnowflakeList(), "XP_WHITELIST")
```
Typo

daspositioned commented on src/main/kotlin/com/github/devcordde/devcordbot/config/codecs/AnimatedGameCodec.kt:
```diff
@@ -0,0 +1,44 @@
+
+        return GameAnimator.AnimatedGame(content, type)
+    }
+}
```
Missing newline

daspositioned commented on src/main/kotlin/com/github/devcordde/devcordbot/database/_Entities.kt:
```diff
@@ -50,15 +49,20 @@ interface DevCordUser {
     }
 
-    override val userID: Long
+    override val userID: Snowflake
```
Can we just change the type without destroying the database?

DRSchlaubi replied:
The underlying type doesn't change we just added a wrapping layer converting the long to a snowflake therefore this shouldn't cause problems. I started the bot on the first ever database we crated for the beta and it worked

daspositioned commented on src/main/kotlin/com/github/devcordde/devcordbot/database/_SnowflakeFields.kt:
```diff
@@ -0,0 +1,66 @@
+ * Shorhand for [EntityClass<Snowflake>][EntityClass].
+ */
+abstract class SnowflakeEntityClass<out E : SnowflakeEntity>(table: IdTable<Snowflake>, entityType: Class<E>? = null) :
+    EntityClass<Snowflake, E>(table, entityType)
```
Missing newline

daspositioned commented on src/main/kotlin/com/github/devcordde/devcordbot/listeners/RankSystemUpdater.kt:
```diff
@@ -150,5 +155,7 @@ enum class Level(
+    LEVEL_100(739064698498187286, 100, LEVEL_75);
+
+    constructor(roleId: Long, level: Int, previousLevel: Level?) : this(Snowflake(roleId), level, previousLevel)
 }
```
Missing newline

daspositioned commented on src/main/kotlin/com/github/devcordde/devcordbot/util/GithubUtil.kt:
```diff
@@ -85,4 +48,5 @@ class GithubUtil(private val client: OkHttpClient) {
  * @property url the url to the contributors GitHub profile
  */
+@Serializable
 data class GitHubContributor(val name: String, val url: String)
```
Missing newline

daspositioned commented on src/main/kotlin/com/github/devcordde/devcordbot/util/HastebinUtil.kt:
```diff
@@ -34,22 +32,18 @@ object HastebinUtil {
 }
+
+@Serializable
+private data class HatebinResponse(
```
Typo

daspositioned commented on src/main/kotlin/com/github/devcordde/devcordbot/util/XPUtil.kt:
```diff
@@ -29,9 +29,9 @@ object XPUtil {
         requiredXp
-    }()
+    }
 }
```
Missing newline

daspositioned commented on src/main/kotlin/de/nycode/bankobot/docdex/CodeBlockFormatter.kt:
```diff
@@ -0,0 +1,64 @@
+            FlexmarkUtils::processPre
+        )).toMutableSet()
+    }
+}
```
Missing newline

DRSchlaubi committed:
Add automatic das_

DRSchlaubi committed:
Port to Kotlin 1.5
- Migrate deprecated API calls
- Target java 16

daspositioned commented on src/main/kotlin/com/github/devcordde/devcordbot/database/_Entities.kt:
```diff
@@ -23,6 +23,7 @@ import org.jetbrains.exposed.dao.Entity
 import org.jetbrains.exposed.dao.EntityClass
 import org.jetbrains.exposed.dao.id.EntityID
 import java.time.Instant
+import java.util.*
```
Why wildcard imports?

DRSchlaubi replied:
Because IntelliJ adds them and they can't be autofixed

daspositioned replied:
It doesn't do that for me but whatever

DRSchlaubi committed:
Apply requested changes

DRSchlaubi committed:
Nvm i was an idiot

daspositioned commented on src/main/kotlin/com/github/devcordde/devcordbot/config/Config.kt:
```diff
@@ -33,7 +33,7 @@ class Config internal constructor(private val config: Konfig) {
      * Key: `XP_WHITELIST`
      */
-    val xpWhielist: List<Snowflake> by config.property(ConfigSpec.xpWhielist)
+    val xpWhielist: List<Snowflake> by config.property(ConfigSpec.xpWhitelist)
```
Typo

DRSchlaubi committed:
Fix another typo

DRSchlaubi commented:
Because there are currently issues with Kord and Kotlin 1.5 I am going to draft this until it got solved

daspositioned committed:
Improve messages

DRSchlaubi committed:
Update libraries to Kotlin 1.5
- Remove GlobalScope calls

DRSchlaubi committed:
Merge remote-tracking branch 'fork/feature/refactor2' into feature/refactor2

DRSchlaubi committed:
Merge remote-tracking branch 'origin/develop' into feature/refactor2

# Conflicts:
#	.github/workflows/gradle.yml
#	Dockerfile
#	build.gradle.kts
#	src/main/kotlin/com/github/devcordde/devcordbot/Launcher.kt
#	src/main/kotlin/com/github/devcordde/devcordbot/core/DevCordBotImpl.kt
#	src/main/kotlin/com/github/devcordde/devcordbot/core/autohelp/ContentFetcher.kt

DRSchlaubi commented:
The Kotlin 1.5 issues have been resolved

DRSchlaubi committed:
run ktlint

daspositioned committed:
Very small message improvement

daspositioned reviewed (dismissed):
Looks pretty decent

DRSchlaubi committed:
Final bugfixes
- Fix bug with database snowflake conversion
- FIx Kotlin 1.5 dependency issues
- Fix DevcordBotImpls CoroutineContext
- Fix Github API
- Use CIO instead of OkHttp for WS
- Fix paginator emote detection
- Don't fire 10 rest requests in /ranks
- Update /tag edit
- Fix recursive modification of level cache

daspositioned reviewed (approved):
Still decent
